src/cmd/build.js

```diff
@@ -40,41 +40,34 @@ export type PackageCreatorParams = {
 export type PackageCreatorFn =
     (params: PackageCreatorParams) => Promise<ExtensionBuildResult>;
 
-function defaultPackageCreator(
+async function defaultPackageCreator(
   {manifestData, sourceDir, fileFilter, artifactsDir}: PackageCreatorParams
 ): Promise<ExtensionBuildResult> {
+  let id;
 
-  return new Promise(
-    (resolve) => {
-      if (manifestData) {
-        const id = getManifestId(manifestData);
-        log.debug(`Using manifest id=${id || '[not specified]'}`);
-        resolve(manifestData);
-      } else {
-        resolve(getValidatedManifest(sourceDir));
-      }
-    })
-    .then((manifestData) => {
-      return zipDir(
-        sourceDir, {
-          filter: (...args) => fileFilter.wantFile(...args),
-        })
-        .then((buffer) => {
-          let packageName = safeFileName(
-            `${manifestData.name}-${manifestData.version}.zip`);
-          let extensionPath = path.join(artifactsDir, packageName);
-          let stream = createWriteStream(extensionPath);
-          let promisedStream = streamToPromise(stream);
-
-          stream.write(buffer, () => stream.end());
-
-          return promisedStream
-            .then(() => {
-              log.info(`Your web extension is ready: ${extensionPath}`);
-              return {extensionPath};
-            });
-        });
-    });
+  if (manifestData) {
+    id = getManifestId(manifestData);
+    log.debug(`Using manifest id=${id || '[not specified]'}`);
+  } else {
+    manifestData = await getValidatedManifest(sourceDir);
+  }
+
+  let buffer =  await zipDir(sourceDir, {
+    filter: (...args) => fileFilter.wantFile(...args),
+  });
+
+  let packageName = safeFileName(
```

Comment:
@pdehaan That's great, I tried to enable the above two eslint rules and let eslint to fix the errors and it seems to be able to fix almost every one of them (besides a small bunch).

Unfortunately these errors are spread all over the repo, and so my proposal is to choose one of the following approaches:

- enable the two eslint rules as warnings instead of errors, fix only the warnings that are into functions (or files) that we are already touching with this pull request and defer the other in a separate followup pull request (to be opened immediately after merging this one)

- defer this part ot the last patch of this pull request, where we enable the two rules as errors and fix all of them, once the async/await part of the change is ready (mainly to make the review process easier on the eyes :-))

@kumar303 how this sounds to you?